workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx

```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
 import catalogIcon from '../../assets/icons/catalog_icon.svg';
 import shieldCheckIcon from '../../assets/icons/shield_check_icon.svg';
 import userIcon from '../../assets/icons/user_icon.svg';
+import {Liferay} from '../../liferay/liferay';
 import {useAppContext} from '../../manage-app-state/AppManageState';
+import {
+	getMyUserAditionalInfos,
+	updateUserAdditionalinfos,
+	updateUserPassword,
+} from '../../utils/api';
+import {createPassword} from '../../utils/createPassword';
 import {Avatar} from '../Avatar/Avatar';
 import {DetailedCard} from '../DetailedCard/DetailedCard';
+import {
+	addAdditionalInfo,
+	callRolesApi,
+	getAccountRolesOnAPI,
+	getSiteURL,
+} from '../InviteMemberModal/services';
 
 interface MemberProfileProps {
-	member: MemberProps;
+	memberUser: MemberProps;
+	renderToast: (message: string, title: string, type: DisplayType) => void;
 	setSelectedMember: (value: MemberProps | undefined) => void;
+	userLogged?: UserLogged;
 }
 
-export function MemberProfile({member, setSelectedMember}: MemberProfileProps) {
+export function MemberProfile({
+	memberUser,
+	renderToast,
+	setSelectedMember,
+	userLogged,
+}: MemberProfileProps) {
 	const [{gravatarAPI}, _] = useAppContext();
 
+	const handlePut = async (event: React.FormEvent) => {
+		event.preventDefault();
+
+		const myUserAdditionalInfos = await getMyUserAditionalInfos(
+			memberUser.userId
+		);
+		const newPassword = createPassword();
+
+		for (const userAdditionInfo of myUserAdditionalInfos.items || []) {
+			const response = await updateUserAdditionalinfos(
```

Comment:
```suggestion
			const updatedUserInfos = await updateUserAdditionalinfos(
```
You can use a more descriptive name for this variable